### PR TITLE
feat(basic-runtime): Setup default entitlements handler to consume metered reads

### PR DESCRIPTION
**Pull request by hartmannr76**
Adds a default `setOnEntitlementsResponse` to the basic runtime that will consume a metered read if an entitlement provided enables the page via metering.

FYI @longmelon @davwood-alpha

### CODE REVIEW

**Comment by jpettitt** [comment]
Shouldn't this live in the publisher code (or swg-basic?)

Did you confirm this won't consume showcase meters?

File: src/runtime/basic-runtime-test.js [hunk 1]
```diff
@@ -186,6 +186,49 @@ describes.realWin('installBasicRuntime', {}, (env) => {
...
+
+    beforeEach(() => {
+      sandbox
+        .stub(BasicRuntime.prototype, 'setOnEntitlementsResponse')
+        .callsFake((callback) => {
+          configuredCallback = callback;
+        });
+      entitlements = {
+        consume: sandbox.stub(),
+        clone: () => entitlements,
```
Comment (ChrisAntaki) [inline src/runtime/basic-runtime-test.js:202]: Could we remove this `clone` method? Looks like the tests never call it, so it's reducing the coverage %
https://app.codecov.io/gh/subscriptions-project/swg-js/compare/2106/diff

**Comment by ChrisAntaki** [comment]
Since this PR changes `basic-runtime.js`, I think the showcase meters should be safe

**Comment by jpettitt** [comment]
@ChrisAntaki if we have a showcase pub using swg-basic what will happen?  Presumably we need to add a bunch of code to support that anyway right?  If so we're good here.

File: src/runtime/basic-runtime-test.js (hunk 1 above)
Comment (hartmannr76) [reply src/runtime/basic-runtime-test.js]: Whoops, yeah. Removed. Had left that before I decided to just stub out and capture the configured callback

**Review by ChrisAntaki — APPROVED** [review]

**Review by jpettitt — APPROVED** [review]

**Pull request merged**